basic02-prog-by-name/xdp_loader.c

```diff
@@ -59,106 +59,76 @@ static const struct option_wrapper long_options[] = {
 };
 
 /* Lesson#1: More advanced load_bpf_object_file and bpf_object */
-struct bpf_object *__load_bpf_object_file(const char *filename, int ifindex)
-{
-	/* In next assignment this will be moved into ../common/ */
-	int first_prog_fd = -1;
-	struct bpf_object *obj;
-	int err;
-
-	/* Lesson#3: This struct allow us to set ifindex, this features is used
-	 * for hardware offloading XDP programs.
-	 */
-	struct bpf_prog_load_attr prog_load_attr = {
-		.prog_type	= BPF_PROG_TYPE_XDP,
-		.ifindex	= ifindex,
-	};
-	prog_load_attr.file = filename;
-
-	/* Use libbpf for extracting BPF byte-code from BPF-ELF object, and
-	 * loading this into the kernel via bpf-syscall
-	 */
-	err = bpf_prog_load_xattr(&prog_load_attr, &obj, &first_prog_fd);
-	if (err) {
-		fprintf(stderr, "ERR: loading BPF-OBJ file(%s) (%d): %s\n",
-			filename, err, strerror(-err));
-		return NULL;
-	}
 
-	/* Notice how a pointer to a libbpf bpf_object is returned */
-	return obj;
-}
 
 /* Lesson#2: This is a central piece of this lesson:
  * - Notice how BPF-ELF obj can have several programs
  * - Find by sec name via: bpf_object__find_program_by_title()
  */
-struct bpf_object *__load_bpf_and_xdp_attach(struct config *cfg)
+struct xdp_program *__load_bpf_and_xdp_attach(struct config *cfg)
 {
 	/* In next assignment this will be moved into ../common/ */
-	struct bpf_program *bpf_prog;
-	struct bpf_object *bpf_obj;
-	int offload_ifindex = 0;
 	int prog_fd = -1;
 	int err;
 
+	DECLARE_LIBBPF_OPTS(bpf_object_open_opts, opts);
+	DECLARE_LIBXDP_OPTS(xdp_program_opts, xdp_opts, 0);
+
+	xdp_opts.open_filename = cfg->filename;
+	xdp_opts.prog_name = cfg->progsec;
+	xdp_opts.opts = &opts;
+
 	/* If flags indicate hardware offload, supply ifindex */
```

Comment:
It looks like the HW offload support for maps is gone from libbpf 1.0.0 https://github.com/libbpf/libbpf/wiki/Libbpf:-the-road-to-v1.0#libbpfh-high-level-apis so we may want to just remove that altogher?